core/store/ledgerstore/ledger_store.go

```diff
@@ -768,7 +815,7 @@ func (this *LedgerStoreImp) executeBlock(block *types.Block) (result store.Execu
 	} else {
 		result.MerkleRoot = this.stateStore.GetStateMerkleRootWithNewHash(result.Hash)
 	}
-
+	this.updateStateToTree(overlay)
```

Comment:
need move to submitBlock .

Reply:
this is used to update state with executing transactions. not commit. do not need to move to submitBlock.

Reply:
I do not find any rollback for stateTree when errors happen between executeBlock and submitBlock.

Reply:
updated

Reply:
fixed